lib/internal/test_runner/reporter/spec.js

```diff
@@ -121,7 +121,10 @@ class SpecReporter extends Transform {
     callback(null, this.#handleEvent({ type, data }));
   }
   _flush(callback) {
-    const results = [`\n${colors['test:fail']}${symbols['test:fail']}failing tests:${white}\n`];
+    const results = [];
+    if (this.#failedTests.length > 0) {
+      ArrayPrototypePush(results, `\n${colors['test:fail']}${symbols['test:fail']}failing tests:${white}\n`);
+    }
```

Comment:
nit: we can skip the array creation if length is 0
```suggestion
    if (this.#failedTests.length === 0) {
      callback (null, '');
      return;
    }
    const results = [`\n${colors['test:fail']}${symbols['test:fail']}failing tests:${white}\n`];
```